internal/api/http.go

```diff
@@ -187,16 +187,21 @@ func inspectableMIMEType(t string) bool {
 	return strings.HasPrefix(t, "text/") || jsonTypeRE.MatchString(t)
 }
 
+func isValidHost(reqHost, hrtHost string) bool {
+	return strings.Contains(reqHost, hrtHost)
```

Comment:
Hostnames in URLs are case-insensitive, so you will want to factor that in when performing comparisons.

To compare whether two values are identical hostnames, you can use `strings.EqualFold(host1, host2)`.

To check whether one hostname is a subdomain of the other, enforce the use of the `.` separator and do strict suffix check. For this, `strings.Contains()` is not a good method because it's imprecise: a host called `thegithub.company` does contain the string `github.com` but is *not* a subdomain of `github.com`. Instead, this is a better check: `strings.HasSuffix("api.github.com", ".github.com")`. Again, remember to normalize the case of input arguments before submitting them for the HasSuffix check.

Reply:
I've made this more robust based on your suggestions.